gqcp/include/QCModel/CC/CCD.hpp

```diff
@@ -0,0 +1,408 @@
+     * 
+     *  @return the F2-intermediate
+     * 
+     *  @note This is one of the intermediate quantities in the factorization of CCD. In particular, F2 represents equation (4) in Stanton1991.
+     */
+    static ImplicitMatrixSlice<Scalar> calculateF2(const QCMatrix<Scalar>& f, const QCRankFourTensor<Scalar>& V_A, const T2Amplitudes<Scalar>& t2) {
+
+        const auto& orbital_space = t2.orbitalSpace();
+
+        // Implement the formula for F2 in equation (4) in Stanton1991.
+        auto F2 = orbital_space.template initializeRepresentableObjectFor<Scalar>(OccupationType::k_occupied, OccupationType::k_occupied);  // zero-initialize an occupied-occupied object
+        for (const auto& m : orbital_space.indices(OccupationType::k_occupied)) {
+            for (const auto& i : orbital_space.indices(OccupationType::k_occupied)) {
+                Scalar value {0.0};  // zero-initialize the scalar value to be added
+
+                // Calculate the contribution from the first term.
+                if (m != i) {  // (1 - delta_mi)
+                    value += f(m, i);
+                }
+
+                // Calculate the contribution from the fourth term.
+                for (const auto& n : orbital_space.indices(OccupationType::k_occupied)) {
+                    for (const auto& e : orbital_space.indices(OccupationType::k_virtual)) {
+                        for (const auto& f : orbital_space.indices(OccupationType::k_virtual)) {
+                            value += 0.5 * t2(i, n, e, f) * V_A(m, n, e, f);
+                        }
+                    }
+                }
+
+                F2(m, i) = value;
+            }
+        }
+
+        return F2;
+    }
+
+
+    /**
+     *  @param f                    the (inactive) Fock matrix
+     *  @param V_A                  the antisymmetrized two-electron integrals (in physicist's notation)
+     *  @param t2                   the T2-amplitudes
+     * 
+     *  @return the F3-intermediate
+     * 
+     *  @note This is one of the intermediate quantities in the factorization of CCD. In particular, F3 represents equation (5) in Stanton1991.
+     */
+    static ImplicitMatrixSlice<Scalar> calculateF3(const QCMatrix<Scalar>& f, const QCRankFourTensor<Scalar>& V_A, const T2Amplitudes<Scalar>& t2) {
+
+        const auto& orbital_space = t2.orbitalSpace();
+
+        // Implement the formula for F3 in equation (5) in Stanton1991.
+        auto F3 = orbital_space.template initializeRepresentableObjectFor<Scalar>(OccupationType::k_occupied, OccupationType::k_virtual);  // zero-initialize an occupied-virtual object
+        for (const auto& m : orbital_space.indices(OccupationType::k_occupied)) {
+            for (const auto& e : orbital_space.indices(OccupationType::k_virtual)) {
+                // Calculate the contribution from the first term.
+                F3(m, e) = f(m, e);
+            }
+        }
+
+        return F3;
+    }
```

Comment:
`F3` seems to be just (a part of) the Fock matrix. Please update the code, avoiding any calculation of the `F3` intermediate.

Reply:
Forgot to adjust this one while adjusting tau2 and tau2_tilde!